extensions/tika/runtime/src/main/java/org/apache/camel/quarkus/component/tika/TikaRecorder.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.tika;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Set;
+
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.tika.TikaContent;
+import io.quarkus.tika.TikaMetadata;
+import io.quarkus.tika.TikaParser;
+import io.quarkus.tika.runtime.TikaParserProducer;
+import org.apache.camel.Component;
+import org.apache.camel.Producer;
+import org.apache.camel.component.tika.TikaComponent;
+import org.apache.camel.component.tika.TikaConfiguration;
+import org.apache.camel.component.tika.TikaEndpoint;
+import org.apache.camel.component.tika.TikaProducer;
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.mime.MediaType;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+
+@Recorder
+public class TikaRecorder {
+
+    public RuntimeValue<TikaComponent> createTikaComponent(BeanContainer container) {
+        return new RuntimeValue<>(new QuarkusTikaComponent(container.instance(TikaParserProducer.class)));
+    }
+
+    @org.apache.camel.spi.annotations.Component("tika")
```

Comment:
Is this annotation required given that we produce a named bean above? https://github.com/apache/camel-quarkus/pull/998/files#diff-6af9bcae1d2af7449582ad99e6bdac3cR54

Reply:
@ppalaga It would make sense to get rid of this annotation. 

But  without it, execution fails for `odf` parser with:

>Caused by: java.lang.LinkageError: loader constraint violation: loader (instance of <bootloader>) previously initiated loading for a different type with name "org/w3c/dom/Node"

I'm not sure about the reason of this behavior. There is a one related comment: https://github.com/quarkusio/quarkus/issues/8375#issuecomment-610309015

Reply:
Strange, that looks like a symptom of some class loading nastyness. Eithere `org/w3c/dom/Node` is coming from two jars or it is loaded by two different class loaders.